File: Assignment__23/Question4.c
#include<stdio.h>
#include<stdlib.h>

void DiaplayInRange(int Arr[], int iSize, int iNo1, int iNo2)
{
    int iCnt = 0;

    for(iCnt = 0; iCnt < iSize; iCnt++)    
    {
        if(Arr[iCnt] > iNo1 && Arr[iCnt] < iNo2)
        {
            printf("%d\n",Arr[iCnt]);
        }
    }
}

int main()
{
    int iLength = 0;
    int iValue1 = 0;
    int iValue2 = 0;
    int iCnt = 0;
    int *ptr = NULL;

    printf("Enter the number of elements : ");
    scanf("%d", &iLength);

    ptr = (int *)malloc(iLength * sizeof(int));
    
    if(ptr == NULL)      
     {
        printf("Unable to allocate the memoey\n");
        return -1;
    }

    printf("Enter the elements : \n");

    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        scanf("%d",&ptr[iCnt]);
    }

    printf("Enter the starting range : \n");
    scanf("%d", &iValue1);

    printf("Enter the the finishing end : \n");
    scanf("%d", &iValue2);

    DiaplayInRange(ptr,iLength,iValue1, iValue2);

    
    
    free(ptr);

    return 0;
}

File: Assignment__23/Question4.java
import java.util.Scanner;

class ArrayX
{
    public int Arr[];
    public int iSize;

    public ArrayX(int iNo)
    {
        iSize = iNo;

        Arr = new int [iSize];    //Need to remember
    }

    public void Accept()
    {
        Scanner sobj1 = new Scanner(System.in);

        int iCnt = 0;

        System.out.println("Enter the numbers : ");

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt] = sobj1.nextInt();
        }
    }

    public void Display()
    {
        System.out.println("The elements of the array are : ");

        int iCnt = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            System.out.println(Arr[iCnt]);
        }

        System.out.println("---------------------------------");
    }

    public void Display(int iNo1, int iNo2)
    {
        int iCnt = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] > iNo1 && Arr[iCnt] < iNo2)
            {
                System.out.println(Arr[iCnt]);
            }   
        }
    }
}

class Question4
{
    public static void main(String A[]) 
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter the number of elements : ");
        int iValue1 = sobj.nextInt();

        System.out.println("Enter the Starting point : ");
        int iValue2 = sobj.nextInt();

        System.out.println("Enter the finishing point : ");
        int iValue3 = sobj.nextInt();

        ArrayX aobj = new ArrayX(iValue1);
        aobj.Accept();
        aobj.Display();

        aobj.Display(iValue2, iValue3);

        aobj = null;
        sobj = null;

        System.gc();
    }
}